### DISCUSSION TITLE
Tree support with infill and 3rd party printers

### DISCUSSION THREAD

**Discussion by jeor112** [OrcaSlicer/OrcaSlicer#79] [Ideas]
Hi
Nice work. Really like this!

I'm working on getting my Tronxy profile to work for this nice Slicer (While I wait for my Bambu to arrive).

It would be very nice to have the infill feature for the Tree and Hybrid support. I have seen this feature in earlier versions.

Secondly, is there a guide on how to add 3rd party printers?

The way I did it was to copy the Voron Trident 350 json files (C:\Program Files\BambuStudio-SoftFever_V1.4.0\resources\profiles) and renamed them to Tronxy and edited the json files.
I have subsequently noticed that if I change something in the files, the changes do not take effect, even not after a restart of the Slicer.
Also, I have noticed, that I only have one filament preset to choose from in the dropdown list while I have selected my Tronxy printer profile.

Thanks in advance

**Comment by erebus04** [comment]
There is a wiki page from Bambu Lab that describes how to create a profile for a third party printer.

https://wiki.bambulab.com/en/software/bambu-studio/add-third-party-printer-in-studio

**Reply by jeor112** [reply]
Thanks, but it's actually the same thing I've done.

First I did it with the demo as described in the guid you are linking to. 
Then I copied the Voron folders and renamed them.
Same results.

Maybe there is something I have missed, but I can't figure out what it is.

**Reply by jeor112** [reply]
A strange thing I have noticed is, that the printer, vendor and filament that I have added, is showing up in the setup wizard.
Just not selectable in the left settings windows.
When I close the program and delete the system folder in C:\Users\XXXX\AppData\Roaming\BambuStudio\system\ and start up again, eg. the new filament I have added shows up in there, but still not selectable in the program.

Same thing happens for both the vendor Slicer as well as in the SoftFever Slicer.

Sorry, I'm not a programming guy and I hope it make any sense for your folks...

**Comment by erebus04** [comment]
Do you have added the compatible Printers in the filamnet.json and the default Filament in the machine.json ?
![filament json](https://user-images.githubusercontent.com/34960970/210086212-4a1bd5e6-f706-4b1e-89f2-d8329322ef08.png)
![machine json](https://user-images.githubusercontent.com/34960970/210086233-4693df0e-cf7e-4b51-bb82-f9d83ccd5f76.png)

**Comment by erebus04** [comment]
Also in the profiles Folder you create the global Json File with the declaration of the Machine, the Filament and the Process ?
![Global Json](https://user-images.githubusercontent.com/34960970/210086407-6530b467-78f3-4b33-a16d-7d0209cb26dc.png)

**Reply by jeor112** [reply]
Yes I have. I will try to upload my settings later. The family is calling 😄

**Comment by SoftFever** [comment]
You are welcome to submit a pull request (PR) with information about the issue you encountered, so that we can review the code and assist you. Please include details about the issue in the PR for us to better understand and address the problem.

**Reply by jeor112** [reply]
I will pretty much like do that, but I'm not sure how to do this in the right way.
I think it would be best, if I can upload a zip file with my settings, because I think there must be something wrong with the structure that I have.
Is that possible?

**Reply by erebus04** [reply]
Yes you can do that.
I look at it and then make a PR

**Reply by jeor112** [reply]
Sorry to ask, but how do I upload a zip file?

**Reply by erebus04** [reply]
Drag and Drop to the Field where you write the Text ;)

**Reply by jeor112** [reply]
Haha, that is easy, said the old dinosaur...
[profiles.zip](https://github.com/SoftFever/BambuStudio-SoftFever/files/10326297/profiles.zip)

Here you go and thank you :-)

**Reply by erebus04** [reply]
There are errors in several files.
I will revise them tomorrow and create a PR.
I will write you then what you have done wrong.

**Reply by jeor112** [reply]
Thank you very much! Sorry for my ignorance. 😳

**Reply by erebus04** [reply]
You are Welcome, no Problem
